src/pgduckdb_types.cpp

```diff
@@ -236,6 +396,16 @@ struct PostgresArrayAppendState {
 		auto &values = duckdb::ListValue::GetChildren(value);
 		idx_t to_append = values.size();
 
+        if (to_append == 0 && dimension < number_of_dimensions-1) {
```

Comment:
This shouldn't be using spaces for indentation. I'm surprised our clang-format check is not failing on this... That's something we should probably look into. Installing editorconfig should help with this too. 

There are a bunch more places in the PR where space indentation is used, instead of tabs.

Reply:
Fixed. But in `.clang-format`, alignment doesn’t have to use tabs, specifically `UseTab: ForIndentation IndentWidth: 4 TabWidth: 4`, where `UseTab` is not `always`. However, in `.editorconfig`, `[*.{c,h,cpp,hpp}] indent_style = tab`. Are these two configurations conflicting? Which one should be used?